File: source/includes/fundamentals/code-snippets/network-compression.rs
use std::env;
use mongodb::{
    bson::doc,
    options::{ ClientOptions, ServerApi, ServerApiVersion, Compressor },
    Client,
};

#[tokio::main]
async fn main() -> mongodb::error::Result<()> {
    // begin-clientoptions
    let uri = "<connection string>";
    let mut client_options = ClientOptions::parse(uri).await?;

    let compressors = vec![
        Compressor::Snappy,
        Compressor::Zstd { level: Some(1) },
        Compressor::Zlib { level: None }
    ];

    client_options.compressors = Some(compressors);
    let client = Client::with_options(client_options)?;
    // end-clientoptions

    client.database("admin").run_command(doc! { "ping": 1 }, None).await?;
    println!("Pinged your deployment. You successfully connected to MongoDB!");

    Ok(())
}

File: content/laravel-mongodb/current/source/includes/connect/network-compression.php
<?php

// start-all-compression-options
'connections' => [
    'mongodb' => [
        'dsn' => 'mongodb+srv://mongodb0.example.com/',
        'driver' => 'mongodb',
        'database' => 'sample_mflix',
        'options' => [
            'compressors' => 'snappy,zstd,zlib',
        ],
    ],
],
// end-all-compression-options

// start-all-compression-uri
'connections' => [
    'mongodb' => [
        'dsn' => 'mongodb+srv://mongodb0.example.com/?compressors=snappy,zstd,zlib',
        'driver' => 'mongodb',
        'database' => 'sample_mflix',
    ],
],
// end-all-compression-uri

// start-zlib-compression-options
'connections' => [
    'mongodb' => [
        'dsn' => 'mongodb+srv://mongodb0.example.com/',
        'driver' => 'mongodb',
        'database' => 'sample_mflix',
        'options' => [
            'compressors' => 'zlib',
            'zlibCompressionLevel' => 1,
        ],
    ],
],
// end-zlib-compression-options

// start-zlib-compression-uri
'connections' => [
    'mongodb' => [
        'dsn' => 'mongodb+srv://mongodb0.example.com/?compressors=zlib&zlibCompressionLevel=1',
        'driver' => 'mongodb',
        'database' => 'sample_mflix',
    ],
],
// end-zlib-compression-uri
